From 3030a34ac12700f5b6ce2d094e4f40508bf7cdde Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Wed, 22 Apr 2015 13:01:34 +0800
Subject: [PATCH 0514/5242] MLK-10713-2 ASoC: imx-sii902x: Add 96k and 192k
 sample rate support

commit  3404e777b4e724a0ab19b61d6c82ac593cfd2b69 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add 96k and 192k sample rate support for hdmi audio.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
(cherry picked from commit fa94b89ab94ce78800d112d0a013d0c0901d9b78)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-sii902x.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-sii902x.c b/sound/soc/fsl/imx-sii902x.c
index 8dfcf4c..72c407d 100644
--- a/sound/soc/fsl/imx-sii902x.c
+++ b/sound/soc/fsl/imx-sii902x.c
@@ -55,8 +55,10 @@ static int imx_sii902x_startup(struct snd_pcm_substream *substream)
 
 	support_rates[0] = 32000;
 	support_rates[1] = 48000;
+	support_rates[2] = 96000;
+	support_rates[3] = 192000;
 	constraint_rates.list = support_rates;
-	constraint_rates.count = 2;
+	constraint_rates.count = 4;
 
 	ret = snd_pcm_hw_constraint_list(runtime, 0, SNDRV_PCM_HW_PARAM_RATE,
 						&constraint_rates);
-- 
1.7.9.5

